homeassistant/components/mcp23017/switch.py

```diff
@@ -0,0 +1,93 @@
+"""Support for switch sensor using I2C MCP23017 chip."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.switch import PLATFORM_SCHEMA
+from homeassistant.const import DEVICE_DEFAULT_NAME
+from homeassistant.helpers.entity import ToggleEntity
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_INVERT_LOGIC = 'invert_logic'
+CONF_I2C_ADDRESS = 'i2c_address'
+CONF_PINS = 'pins'
+CONF_PULL_MODE = 'pull_mode'
+
+DEFAULT_INVERT_LOGIC = False
+DEFAULT_I2C_ADDRESS = 0x20
+
+_SWITCHES_SCHEMA = vol.Schema({
+    cv.positive_int: cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PINS): _SWITCHES_SCHEMA,
+    vol.Optional(CONF_INVERT_LOGIC, default=DEFAULT_INVERT_LOGIC): cv.boolean,
+    vol.Optional(CONF_I2C_ADDRESS, default=DEFAULT_I2C_ADDRESS):
+    vol.Coerce(int),
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the MCP23017 devices."""
+    import board
+    import busio
+    import adafruit_mcp230xx
+
+    invert_logic = config.get(CONF_INVERT_LOGIC)
+    i2c_address = config.get(CONF_I2C_ADDRESS)
+
+    i2c = busio.I2C(board.SCL, board.SDA)
+    mcp = adafruit_mcp230xx.MCP23017(i2c, address=i2c_address)
+
+    switches = []
+    pins = config.get(CONF_PINS)
+    for pin_num, pin_name in pins.items():
+        pin = mcp.get_pin(pin_num)
+        switches.append(MCP23017Switch(pin_name, pin, invert_logic))
+    async_add_entities(switches)
+
+
+class MCP23017Switch(ToggleEntity):
+    """Representation of a  MCP23017 output pin."""
+
+    def __init__(self, name, pin, invert_logic):
+        """Initialize the pin."""
+        import digitalio
+        self._name = name or DEVICE_DEFAULT_NAME
+        self._pin = pin
+        self._invert_logic = invert_logic
+        self._state = False
```

Comment:
So that means that restarting Home Assistant will change the state. Is that a problem?

Reply:
Well, we have to initialize the pin to some value. I based the implementation in what rpi_gpio initializes the output pins to.
We could create a config option for that, something like: ```initial_state```.

Reply:
> Well, we have to initialize the pin to some value.

I guess I don't understand that ... what would happen if we did not?

Just asking for my own education. Let's keep it as it is.

Reply:
Now that you mention it, we could read the current pin state after setting it up as output and assign that to the switch state like ```self._state = self._pin.value```. Assuming that the pin wasn't changed through another program it should give us the last pin status before restarting HA, and it would initialize on first start to the default value which should be LOW (False).
I think this is a better approach... thanks for pointing it out.

Reply:
I think I talked to soon because that brings an issue if you use ```inverse_logic: True``` to run, let's say, and active low relay. In which case on first start the relay will be turned on.
I think the safest is still to initialize it to False on reboot.